config_file_type.go

```diff
@@ -57,21 +58,58 @@ func (f *ConfigFileType) Type() string {
 
 // Stat checks if the file exists and computes the platform specific Path and
 // directly writes to the provided diagnostics.
-func (f *ConfigFileType) Stat(diags *diag.Diagnostics, component diag.Component, cfg *Config, dirPath string) bool {
-	for _, ext := range f.Extensions {
-		cfgFilePath := dirPath + string(filepath.Separator) + cfg.FileName + "." + ext
+func (f *ConfigFileType) Stat(diags *diag.Diagnostics, component diag.Component, cfg *Config, filePath string) bool {
+	// todo: tidy `ConfigFileType.Stat` implementation. there should be a better way.
```

Comment:
yeah, not a fan of this.
feels like we're trying to do two different things via the same func, but it's not cohesive.

Reply:
TODO: consider later. functional.